### DISCUSSION TITLE
What is the process of a new release?

### DISCUSSION THREAD

**Discussion by karmingc** [pyenv/pyenv#2313] [Q&A]
More specifically, i'm interested in using python 3.10.4 ([merged PR](https://github.com/pyenv/pyenv/pull/2295)) through pyenv. How frequent are releases in general?

**Comment by edgarrmondragon** [comment]
@karmingc you can install 3.10.4 with pyenv today by using a definition file that you can download from https://github.com/pyenv/pyenv/blob/6d20b3b12383084e7f3c919947363dfe7ef33916/plugins/python-build/share/python-build/3.10.4.

Example:

```console
$ cat 3.10.4
#require_gcc
prefer_openssl11
export PYTHON_BUILD_CONFIGURE_WITH_OPENSSL=1
install_package "openssl-1.1.1n" "https://www.openssl.org/source/openssl-1.1.1n.tar.gz#40dceb51a4f6a5275bde0e6bf20ef4b91bfc32ed57c0552e2e8e15463372b17a" mac_openssl --if has_broken_mac_openssl
install_package "readline-8.0" "https://ftpmirror.gnu.org/readline/readline-8.0.tar.gz#e339f51971478d369f8a053a330a190781acb9864cf4c541060f12078948e461" mac_readline --if has_broken_mac_readline
if has_tar_xz_support; then
    install_package "Python-3.10.4" "https://www.python.org/ftp/python/3.10.4/Python-3.10.4.tar.xz#80bf925f571da436b35210886cf79f6eb5fa5d6c571316b73568343451f77a19" standard verify_py310 copy_python_gdb ensurepip
else
    install_package "Python-3.10.4" "https://www.python.org/ftp/python/3.10.4/Python-3.10.4.tgz#f3bcc65b1d5f1dc78675c746c98fcee823c038168fc629c5935b044d0911ad28" standard verify_py310 copy_python_gdb ensurepip
fi

$ pyenv versions | grep 3.10
3.10.1
3.10.2
3.10.3

$ pyenv install 3.10.4  # <- 3.10.4 is the name of the file, not the upstream version
python-build: use openssl@1.1 from homebrew
python-build: use readline from homebrew
Downloading Python-3.10.4.tar.xz...
-> https://www.python.org/ftp/python/3.10.4/Python-3.10.4.tar.xz
Installing Python-3.10.4...
python-build: use tcl-tk from homebrew
python-build: use readline from homebrew
python-build: use zlib from xcode sdk
Installed Python-3.10.4 to /Users/imauser/.pyenv/versions/3.10.4

$ pyenv versions | grep 3.10
3.10.1
3.10.2
3.10.3
3.10.4

**Comment by ulgens** [comment]
After every CPython release, dependabot and brew users wait until pyenv's new release to be able to use that CPython release. And almost every time, it gets released with 2-3 weeks of delay. So instead of making users wait, is it possible to tie pyenv releases with CPython?

**Reply by edgarrmondragon** [reply]
@ulgens (Disclaimer: I'm not a maintainer of pyenv).

I'm not following.

1. Why would usage of dependabot be blocked by a release of pyenv?

2. With the example above, you don't need to wait for a new release (and thus a new version being available to brew) to install the most recent CPython, even if it not's been added to to the pyenv repo!

> is it possible to tie pyenv releases with CPython

Having worked on similarly upstream-dependent projects, tying releases is usually a non-trivial effort

**Reply by native-api** [reply]
CPython releases come quite often -- almost every week and sometimes multiple releases in a week (see e.g. the dates at https://www.python.org/ftp/python/).
Making new Pyenv releases so often would be rather taxing.
Currently, we make a new release roughly once a month and allow to install the development head for those who always chase the "latest and greatest" (in Homebrew, you can `brew install pyenv --head`).

**Reply by native-api** [reply]
I guess that would be realistic if we automate the process.

**Reply by anton-petrov** [reply]
We can use something like this https://github.com/marketplace/actions/automatic-releases, to automate our releases.

**Reply by ulgens** [reply]
@edgarrmondragon 

1. It's not blocked, but dependabot developers prefer to use release versions in their code (It's been the case for the last couple of years. On multiple occasions, I was the one who pushed new Python and pyenv versions to dependabot repo) Considering dependabot is bought by Github and the functionality it provides is ready to use for any existing repo, I consider them as a significant user of pyenv.
2. That's true, someone can install any CPython release without needing pyenv at all but also irrelevant IMO. We are discussing this here because we choose to install them with pyenv.

@native-api Thanks for the `--head` tip.

@native-api @anton-petrov +1 for the automation idea.

**Reply by Ninzero** [reply]
Perhaps we could separate Pyenv and Python versions, like Homebrew and its taps. Pyenv release will only focus on functional update and bug fixes, not new versions.
